game-core/src/main/java/games/strategy/triplea/settings/ProtectedStringClientSetting.java

```diff
@@ -0,0 +1,51 @@
+package games.strategy.triplea.settings;
+
+import java.util.Arrays;
+
+import games.strategy.security.CredentialManager;
+import games.strategy.security.CredentialManagerException;
+
+/**
+ * ClientSetting to store encrypted versions of potentially sensitive Strings.
+ */
+class ProtectedStringClientSetting extends ClientSetting<char[]> {
+
+  private final boolean sensitive;
+
+  ProtectedStringClientSetting(final String name, final boolean sensitive) {
+    super(char[].class, name);
+    this.sensitive = sensitive;
+  }
+
+  /**
+   * Protects the given char array and cleans the content afterwards.
+   */
+  @Override
+  protected String formatValue(final char[] value) {
+    try (CredentialManager manager = CredentialManager.newInstance()) {
+      return manager.protect(value);
+    } catch (final CredentialManagerException e) {
+      throw new IllegalStateException("Error while trying to protect String.", e);
+    } finally {
+      Arrays.fill(value, '\0');
```

Comment:
I suspect this is why you had to call `password.clone()` within `SelectionComponentFactory#emailSettings()`.

Reply:
Yeah, unfortunately we can't control what happens to the array after it lands in the map that gets returned.
The UI can't and shouldn't know about any special handling.
The only thing we know for certain is that at one point the array gets passed to setValue so handling it there would be the most obvious thing to do.

Reply:
But when thinking about this a little bit more I think the abstraction with the clone is uneccessary and can be simplified.

Reply:
As an example of the kind of problem this may cause...  I spent twenty minutes last night trying to figure out why `SaveHandler` wasn't displaying the correct new username while I was playing around with moving the "display value" logic to the UI layer.  It turns out I was using the map value after `GameSetting#setObjectValue()` had been called, so the array had been cleared by the time I went to display its value.  Not fun. :disappointed: 

I think this violates the Principle of Least Surprise.  If there were a way to document it, that might be okay.  But because scrubbing the array is an implementation detail of this class, and not some part of the interface contract, trying to put it in there could leave to surprising issues down the road.

Reply:
I don't really have a better option right now. 
I'd do some changes in the next hour (will be away after that) if I can make it in time and if you want to spend time into redoing the code which affects other properties as well, you could do that in a follow-up PR if that's OK for you.
I'm not going to fiddle around with the code the next couple days then

Reply:
Yeah, no problem.  I can take a look at it next week, if needed.

Reply:
Are we clearing out an array for security reasons? This was previously recommended but is no longer, it's security theater. There is no control of when the array overwrite will happen, if  at all. The compiler could decide to optimize the dead store away. Secondly, the window of vulnerability for reading live memory is still there, reducing it does not close the vulnerability.